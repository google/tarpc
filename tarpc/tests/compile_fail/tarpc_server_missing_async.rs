#[tarpc::service(derive_serde = false)]
trait World {
    async fn hello(name: String) -> String;
}

struct HelloServer;

#[tarpc::server]
impl World for HelloServer {
    fn hello(name: String) -> String {
        format!("Hello, {name}!", name)
    }
}

fn main() {}
