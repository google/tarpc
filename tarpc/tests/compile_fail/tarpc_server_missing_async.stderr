error: not all trait items implemented, missing: `HelloFut`
 --> $DIR/tarpc_server_missing_async.rs:9:1
  |
9 | impl World for HelloServer {
  | ^^^^

error: hint: `#[tarpc::server]` only rewrites async fns, and `fn hello` is not async
  --> $DIR/tarpc_server_missing_async.rs:10:5
   |
10 |     fn hello(name: String) ->  String {
   |     ^^
